core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -3163,4 +3382,459 @@ private static List<HTTPRequest> allRequests(RESTCatalogAdapter adapter) {
+    Table table = createTableWithScanPlanning("iterator_close_table");
+
+    // Ensure we have a RESTTable with server-side planning enabled
+    assertThat(table).isInstanceOf(RESTTable.class);
+    RESTTable restTable = (RESTTable) table;
+
+    TableScan scan = restTable.newScan();
+    assertThat(scan).isInstanceOf(RESTTableScan.class);
+    boolean cancelled = isCancelled((RESTTableScan) scan);
+    assertThat(cancelled).isFalse(); // No active plan to cancel
+  }
+
+  private static boolean isCancelled(RESTTableScan scan) throws IOException {
+
+    // Get the iterable and iterator
+    CloseableIterable<FileScanTask> iterable = scan.planFiles();
+    CloseableIterator<FileScanTask> iterator = iterable.iterator();
+
+    // Verify we can close the iterator without exceptions
+    // The cancellation callback will be called (though no active plan exists)
+    iterator.close();
+
+    // Verify we can still call cancelPlan on the scan
+    return scan.cancelPlan();
+  }
+
+  @Test
+  public void testMetadataTablesWithRemotePlanning() throws IOException {
+    configurePlanningBehavior(TestPlanningBehavior.Builder::synchronousWithAllTasks);
+    Table table = createTableWithScanPlanning("metadata_tables_test");
+
+    // Ensure we have a RESTTable with server-side planning enabled
+    assertThat(table).isInstanceOf(RESTTable.class);
+
+    // Test that metadata tables work with remote planning
+    // Files metadata table
+    Table filesTable =
+        MetadataTableUtils.createMetadataTableInstance(table, MetadataTableType.FILES);
+    assertThat(filesTable).isNotNull();
+
+    TableScan filesScan = filesTable.newScan();
+    assertThat(filesScan).isNotNull();
+
+    // Verify metadata table scan works (should not use REST scan planning)
+    CloseableIterable<FileScanTask> filesIterable = filesScan.planFiles();
+    List<FileScanTask> filesTasks = Lists.newArrayList(filesIterable);
+    assertThat(filesTasks).isNotEmpty();
+
+    // Snapshots metadata table
+    Table snapshotsTable =
+        MetadataTableUtils.createMetadataTableInstance(table, MetadataTableType.SNAPSHOTS);
+    assertThat(snapshotsTable).isNotNull();
+
+    TableScan snapshotsScan = snapshotsTable.newScan();
+    CloseableIterable<FileScanTask> snapshotsIterable = snapshotsScan.planFiles();
+    List<FileScanTask> snapshotsTasks = Lists.newArrayList(snapshotsIterable);
+    assertThat(snapshotsTasks).isNotEmpty();
+
+    // Manifests metadata table
+    Table manifestsTable =
+        MetadataTableUtils.createMetadataTableInstance(table, MetadataTableType.MANIFESTS);
+    assertThat(manifestsTable).isNotNull();
+
+    TableScan manifestsScan = manifestsTable.newScan();
+    CloseableIterable<FileScanTask> manifestsIterable = manifestsScan.planFiles();
+    List<FileScanTask> manifestsTasks = Lists.newArrayList(manifestsIterable);
+    assertThat(manifestsTasks).isNotEmpty();
+  }
+
+  @Test
+  public void testIterableCloseTriggersCancel() throws IOException {
```

Comment:
this sounds like something that should be tested in a `TestScanTaskIterable`. Same for `testIteratorCloseTriggersCancel`